Sources/TuistSigning/SigningCipher.swift

```diff
@@ -38,46 +38,134 @@ enum SigningCipherError: FatalError, Equatable {
+        let masterKey = try self.masterKey(at: path)
+        let signingKeyFiles = try locateUnencryptedSigningFiles(at: path)
+        guard !signingKeyFiles.isEmpty else { return }
+
+        let correctlyEncryptedSigningFiles = try self.correctlyEncryptedSigningFiles(at: path, masterKey: masterKey)
+
+        try locateEncryptedSigningFiles(at: path)
+            .filter { !correctlyEncryptedSigningFiles.map(\.encrypted).contains($0) }
+            .forEach(FileHandler.shared.delete)
+        
         let cipheredKeys = try signingKeyFiles
+            .filter { !correctlyEncryptedSigningFiles.map(\.unencrypted).contains($0) }
             .map(FileHandler.shared.readFile)
             .map { try encryptData($0, masterKey: masterKey) }
 
-        try zip(cipheredKeys, signingKeyFiles).forEach {
-            logger.debug("Encrypting \($1.pathString)")
-            try $0.write(to: $1.url)
+        try zip(cipheredKeys, signingKeyFiles)
+            .forEach { key, file in
+                logger.debug("Encrypting \(file.pathString)")
+                let encryptedPath = AbsolutePath(file.pathString + "." + Constants.encryptedExtension)
+                try key.write(to: encryptedPath.url)
+            }
+
+        if !keepFiles {
+            try signingKeyFiles.forEach(FileHandler.shared.delete)
         }
     }
 
-    public func decryptSigning(at path: AbsolutePath) throws {
-        let (signingKeyFiles, masterKey) = try signingData(at: path)
+    public func decryptSigning(at path: AbsolutePath, keepFiles: Bool) throws {
+        let masterKey = try self.masterKey(at: path)
+        let signingKeyFiles = try locateEncryptedSigningFiles(at: path)
+        guard !signingKeyFiles.isEmpty else { return }
         let decipheredKeys = try signingKeyFiles
             .map(FileHandler.shared.readFile)
             .map {
                 try decryptData($0, masterKey: masterKey)
             }
 
+        try locateUnencryptedSigningFiles(at: path)
+            .forEach(FileHandler.shared.delete)
+
         try zip(decipheredKeys, signingKeyFiles).forEach {
             logger.debug("Decrypting \($1.pathString)")
-            try $0.write(to: $1.url)
+            let decryptedPath = AbsolutePath($1.parentDirectory.pathString + "/" + $1.basenameWithoutExt)
+            try $0.write(to: decryptedPath.url)
+        }
+        
+        if !keepFiles {
+            try signingKeyFiles.forEach(FileHandler.shared.delete)
         }
     }
+    
+    /// - Returns: Master key data
+    public func readMasterKey(at path: AbsolutePath) throws -> String {
+        guard
+            let rootDirectory = rootDirectoryLocator.locate(from: path)
+        else { throw SigningCipherError.signingDirectoryNotFound(path) }
+        let masterKeyFile = rootDirectory.appending(components: Constants.tuistDirectoryName, Constants.masterKey)
+        guard FileHandler.shared.exists(masterKeyFile) else { throw SigningCipherError.masterKeyNotFound(masterKeyFile) }
```

Comment:
If the master key doesn't exist I'd generate one and let the developer know:

```
master.key not found at xxxx. Generating one and adding it to the .gitignore
```

Reply:
I'd agree, but it is a little bit more complicated than that. Let's say I git clone a repository that has encrypted files at `Tuist/Signing`. If I then generate `master.key`, the decryption will fail later on and at that point it might not be so obvious to the user what went wrong (and we can not tell either - besides the fact that they a wrong `master.key`).

I think the best behavior would be as follows:
- if there is encrypted content in the directory, do not generate `master.key` and add a warning (or error?) about missing `master.key`
- it there is *not* encrypted content, simply generate a new `master.key`

Does this make sense?

Reply:
> Does this make sense?

It does. That way we tell the user if some needs to give them access to the key, or if we had to generate a key for them because it did not exist before.